### PR TITLE
Cr client merge

### PR DESCRIPTION
**Pull request by cents02**
https://github.com/AttorneyOnline/AO2-Client/pull/130 knock off PR

### CODE REVIEW

**Comment by oldmud0** [comment]
You may wish to build your branch on your machine after pulling new changes from master:
```
src/aocharmovie.cpp:45:24: error: use of undeclared identifier 'alt_path_still'
  else if (file_exists(alt_path_still))
                       ^
src/aocharmovie.cpp:46:16: error: use of undeclared identifier 'alt_path_still'
    gif_path = alt_path_still;
```

**Comment by cents02** [comment]
@oldmud0 can you 're trigger the CI? It builds on my machine

**Comment by stonedDiscord** [comment]
https://github.com/AttorneyOnline/AO2-Client/blob/78153bcf92f13584d30f2019f0f267ad07b2eae3/src/courtroom.cpp#L1882

you forgot a bracket }

**Comment by stonedDiscord** [comment]
rest in peace audio device picker

File: src/courtroom.cpp [hunk 1]
```diff
@@ -2596,11 +2596,22 @@ void Courtroom::append_ic_text(QString p_text, QString p_name,
...
+          first_message_sent = true;
+        }
+      }
+      else if (force_filter || is_songchange || !mirror_iclog) {
+        if (mirror_iclog) {
+          ui_ic_chatlog->textCursor().insertText('\n' + p_name, bold);
+        }
+        else {
+          ui_ic_chatlog->textCursor().insertText(p_name, bold);
+        }
```
Comment (oldmud0) [inline src/courtroom.cpp]: What?

**Comment by likeawindrammer** [comment]
[backtrace.txt](https://github.com/AttorneyOnline/AO2-Client/files/4465942/backtrace.txt)
Had this crash while trying to test the fix for the *random crash on loading*

**Comment by likeawindrammer** [comment]
[new bug.txt](https://github.com/AttorneyOnline/AO2-Client/files/4465958/new.bug.txt)
Another bug, happened while writing on the IC chat.

**Comment by cents02** [comment]
@likeawindrammer can you be more specific on what was happening at the time which caused the crash?

**Comment by likeawindrammer** [comment]
I'll see if I can replicate it and record it on video. Like I said it happened while writing on the IC chat. And I didn't clic the back to lobby button.

**Comment by likeawindrammer** [comment]
Captura de pantalla de 2020-04-12 14-47-18
This is probably related to the first backtrace. The area list is out of order. 'Lobby' should be first for example.

**Comment by oldmud0** [comment]
Last call for bugs to report.

**Comment by likeawindrammer** [comment]
Give me a few hours to test it a bit more to see if I can find anything.

**Comment by likeawindrammer** [comment]
First as expected there are some new strings to translate added on this branch, but updating the translations could easily be done after the merge.

Second there is a problem with the portuguese translation on this branch that doesn't exist on master:
Captura de pantalla de 2020-04-14 23-04-04

File: src/aooptionsdialog.cpp [hunk 2]
```diff
@@ -1,596 +1,743 @@
...
+  ui_gameplay_form->setWidget(11, QFormLayout::FieldRole, ui_epilepsy_cb);
+
+  ui_language_combobox = new QComboBox(ui_form_layout_widget);
+  ui_language_combobox->addItem(
+      configini->value("language", "  ").value<QString>() +
+      " - Keep current setting");
+  ui_language_combobox->addItem("   - Default");
+  ui_language_combobox->addItem("en - English");
+  ui_language_combobox->addItem("de - Deutsch");
+  ui_language_combobox->addItem("es - Español");
```
Comment (likeawindrammer) [inline src/aooptionsdialog.cpp:192]: Spanish option added twice and portuguese missing, see https://github.com/AttorneyOnline/AO2-Client/blob/901012eb2872ad5d4af23b82cc6dc0d1da587b20/src/aooptionsdialog.cpp#L191

**Comment by likeawindrammer** [comment]
I don't understand what are these for. Also "Colorful IC log:" is missing a tooltip.
Captura de pantalla de 2020-04-14 23-13-06

**Comment by likeawindrammer** [comment]
Nice, I couldn't reproduce any previous bugs, I assume they are gone for good.
My only last complaint is about changing translations.
Captura de pantalla de 2020-04-15 14-35-46
The languages dropdown and themes dropdown don't have a text label to indicate what are they for. And the checkbox on "language:" seems redundant.

**Comment by oldmud0** [comment]
Seems like form elements got jumbled up. The check box probably belongs to something else.

**Comment by cents02** [comment]
For logging purposes incases somebody is looking back at this PR:
Issues presented by windrammer were resolved:
 - The area/character order issue was resolved by removing the qrunnables and turning them into ordinary functions in order to avoid the race condition issue
- The settings were fixed by reverting the code for gameplay settings back to master.

I don't think think there are any other bugs, so unless @likeawindrammer or @oldmud0 have any other concerns I believe the PR should be merged.

**Comment by likeawindrammer** [comment]
I couldn't find anything else, I'll say it's good for merge.

**Comment by oldmud0** [comment]
All right, I'll merge it tomorrow and get the release machinery ready.

**Comment by likeawindrammer** [comment]
@oldmud0 After CR is merged I would like to update the Spanish and Portuguese translations and then do a more broad "beta testing" with people from my servers to make sure both code and translations are perfect before the release. I won't take long, I'll get into updating the translations as soon as I finish the launcher on website thing.

**Comment by oldmud0** [comment]
This branch is pretty complicated. 140 commits with various internal merges, complicated merge conflict resolutions, and aptly named commit messages. I can't do an interactive rebase due to these internal merges, but I will take the final changes and split them into two commits:

- Formatting changes made by `clang-format`
- Actual additions

...which has been done in [2.7-rc](https://github.com/AttorneyOnline/AO2-Client/tree/2.7-rc). Took me about 2 hours to split the changes into two commits. I hope I don't have to do anything like this again.

We're done with the review process here. I will merge the squashed changes with master as soon as they pass CI.

**Pull request closed without merging**